libfs/file_info.go

```diff
@@ -65,7 +66,51 @@ func (fi *FileInfo) IsDir() bool {
 	return fi.ei.Type == libkbfs.Dir
 }
 
+// LastWriterGetter is an interface for something that can return the
+// last KBFS writer of a directory entry.
+type LastWriterGetter interface {
+	LastWriter() (keybase1.User, error)
+}
+
+type fileInfoSys struct {
+	fi *FileInfo
+}
+
+var _ LastWriterGetter = fileInfoSys{}
+
+func (fis fileInfoSys) LastWriter() (keybase1.User, error) {
+	if fis.fi.node == nil {
+		// This won't return any last writer for symlinks themselves.
+		// TODO: if we want symlink last writers, we'll need to add a
+		// new interface to KBFSOps to get them.
+		return keybase1.User{}, nil
+	}
+	md, err := fis.fi.fs.config.KBFSOps().GetNodeMetadata(
+		fis.fi.fs.ctx, fis.fi.node)
+	if err != nil {
+		return keybase1.User{}, err
+	}
+	lastWriterName := md.LastWriterUnverified
```

Comment:
Is there a way to get verified writer, or would that be too much plumbing?

Reply:
> Is there a way to get verified writer, or would that be too much plumbing?

It would require doing something with the Merkle tree and the MD history that we haven't fully figured out yet.  There's no easy way to map a particular file back to its last-modified MD revision.